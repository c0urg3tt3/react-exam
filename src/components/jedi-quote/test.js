import React from 'react'

import { shallow } from 'enzyme'

import JediQuote from './component'

describe('[component] JediQuote', () => {
  it('should render component', () => {
    const props = {
      message: "Something went wrong while fetching !",
      quote: "What if I told you that the Republic was now under the control of a Dark Lord of the Sith?",
      autor: "Count Dooku"
    }

    const wrapper = shallow(<JediQuote/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('div')).toEqual(true)
    expect(wrapper.hasClass('jedi-quote')).toEqual(true)
    expect(wrapper.find('p')).toHaveLength(2)
    expect(wrapper.find('blockquote')).toHaveLength(1)
    expect(wrapper.find('footer')).toHaveLength(1)
    expect(wrapper.find('p').first().text()).toEqual("")
    expect(wrapper.find('blockquote p').first().text()).toEqual("")
    expect(wrapper.find('footer').first().text()).toEqual("")
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps(props)

    expect(wrapper.find('p').first().text()).toEqual(props.message)
    expect(wrapper.find('blockquote p').first().text()).toEqual(props.quote)
    expect(wrapper.find('footer').first().text()).toEqual(props.autor)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
