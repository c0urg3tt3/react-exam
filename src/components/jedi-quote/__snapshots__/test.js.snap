// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediQuote should render component 1`] = `"<div class=\\"jedi-quote\\"><p></p><blockquote><p></p><footer></footer></blockquote></div>"`;

exports[`[component] JediQuote should render component 2`] = `"<div class=\\"jedi-quote\\"><p>Something went wrong while fetching !</p><blockquote><p>What if I told you that the Republic was now under the control of a Dark Lord of the Sith?</p><footer>Count Dooku</footer></blockquote></div>"`;
